### PR TITLE
chore: bump go-header and go-fraud

### PR DESCRIPTION
**Pull request by Wondertan**
This PR bumps go-header and go-fraud which are both breaking - lots of changes here so please review carefully.

Note that this also contains re-introduction of VerifyCommitLightTrusting which caused us issues in `mocha` due to rapid voting power changes. At the moment, there are no test utilities to test that specific case, so we need to wait to roll it out to `mocha`. 

Related: https://github.com/celestiaorg/celestia-app/issues/2382

### CODE REVIEW

File: das/daser_test.go [hunk 1]
```diff
@@ -229,15 +245,15 @@ func TestDASerSampleTimeout(t *testing.T) {
 // createDASerSubcomponents takes numGetter (number of headers
 // to store in mockGetter) and numSub (number of headers to store
 // in the mock header.Subscriber), returning a newly instantiated
-// mockGetter, share.Availability, and mock header.Subscriber.
+// mockGetter, resetshare.Availability, and mock header.Subscriber.
```
Comment (distractedm1nd) [inline das/daser_test.go:248]: accident?

File: state/core_access.go [hunk 2]
```diff
@@ -294,7 +295,7 @@ func (ca *CoreAccessor) BalanceForAddress(ctx context.Context, addr Address) (*B
 	abciReq := abci.RequestQuery{
 		// TODO @renayay: once https://github.com/cosmos/cosmos-sdk/pull/12674 is merged, use const instead
 		Path:   fmt.Sprintf("store/%s/key", banktypes.StoreKey),
-		Height: head.Height() - 1,
+		Height: int64(head.Height() - 1), // TODO(@Wondertan): Why -1?
```
Comment (renaynay) [inline state/core_access.go:298]: n-1 problem :)

File: header/header.go [hunk 3]
```diff
@@ -169,6 +166,41 @@ func (eh *ExtendedHeader) Validate() error {
...
+				),
+			}
+		}
+
+		return nil
+	}
+
+	if err := eh.ValidatorSet.VerifyCommitLightTrusting(eh.ChainID(), untrst.Commit, light.DefaultTrustLevel); err != nil {
+		return &libhead.VerifyError{
+			Reason: err,
```
[the remark this answers is not in the document]
Comment (Wondertan) [reply header/header.go:198]: Great catch❤️

File: header/serde.go [hunk 4]
```diff
@@ -61,7 +61,7 @@ func UnmarshalExtendedHeader(data []byte) (*ExtendedHeader, error) {
 		return nil, err
 	}
 
-	return out, out.Validate()
+	return out, nil
```
[the remark this answers is not in the document]
Comment (Wondertan) [reply header/serde.go:64]: Yes, it's now automatically called

File: state/core_access.go (hunk 2 above)
Comment (Wondertan) [reply state/core_access.go:298]: Ah right, could you document ?

File: nodebuilder/fraud/constructors.go [hunk 5]
```diff
@@ -16,32 +14,46 @@ import (
...
-		})
-		return &Service{
-			Service: pservice,
-		}, pservice, nil
-	}
+func fraudUnmarshaler() fraud.ProofUnmarshaler[*header.ExtendedHeader] {
+	return defaultProofUnmarshaler
+}
+
+func newFraudServiceWithSync(
```
[the remark this answers is not in the document]
Comment (renaynay) [reply nodebuilder/fraud/constructors.go:21]: idc

File: das/daser_test.go (hunk 1 above)
Comment (renaynay) [reply das/daser_test.go]: fixing

File: nodebuilder/fraud/constructors.go [hunk 6]
```diff
@@ -16,32 +14,46 @@ import (
...
+	lc fx.Lifecycle,
+	sub *pubsub.PubSub,
+	host host.Host,
+	hstore libhead.Store[*header.ExtendedHeader],
+	registry fraud.ProofUnmarshaler[*header.ExtendedHeader],
+	ds datastore.Batching,
+	network p2p.Network,
+) (Module, fraud.Service[*header.ExtendedHeader], error) {
+	syncerEnabled := true
```
Comment (vgonkivs) [inline nodebuilder/fraud/constructors.go:30]: do not need this variable anymore as we now have two separate functions

File: nodebuilder/fraud/constructors.go [hunk 7]
```diff
@@ -16,32 +14,46 @@ import (
...
+func newFraudServiceWithoutSync(
+	lc fx.Lifecycle,
+	sub *pubsub.PubSub,
+	host host.Host,
+	hstore libhead.Store[*header.ExtendedHeader],
+	registry fraud.ProofUnmarshaler[*header.ExtendedHeader],
+	ds datastore.Batching,
+	network p2p.Network,
+) (Module, fraud.Service[*header.ExtendedHeader], error) {
+	syncerEnabled := false
```
Comment (vgonkivs) [inline nodebuilder/fraud/constructors.go:50]: the same

**Review by vgonkivs — CHANGES_REQUESTED** [review]

File: header/header.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (vgonkivs) [reply header/header.go]: Why should it be soft though?

File: nodebuilder/fraud/fraud.go [hunk 8]
```diff
@@ -35,3 +39,83 @@ func (api *API) Subscribe(ctx context.Context, proofType fraud.ProofType) (<-cha
 func (api *API) Get(ctx context.Context, proofType fraud.ProofType) ([]Proof, error) {
 	return api.Internal.Get(ctx, proofType)
 }
+
+var _ Module = (*module)(nil)
+
+// module is an implementation of Module that uses fraud.module as a backend. It is used to
```
Comment (distractedm1nd) [inline nodebuilder/fraud/fraud.go:45]:
```suggestion
// module is an implementation of Module that uses fraud.Module as a backend. It is used to
```

:)

**Review by distractedm1nd — APPROVED** [review]
utACK

File: nodebuilder/fraud/fraud.go (hunk 8 above)
Comment (vgonkivs) [reply nodebuilder/fraud/fraud.go:45]: 🚔

File: nodebuilder/fraud/constructors.go (hunk 6 above)
Comment (renaynay) [reply nodebuilder/fraud/constructors.go:30]: it's local to the function, it's too much of a nit to address

**Review by renaynay — APPROVED** [review]

**Review by vgonkivs — APPROVED** [review]

**Pull request merged**